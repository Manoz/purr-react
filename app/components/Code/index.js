import styled from 'styled-components';

const Code = styled.code`
  padding: .2em .4em;
  margin: 0;
  font-weight: 400;
  font-size: 88.5%;
  color: black;
  background-color: rgba(27, 31, 35, .05);
  border-radius: 3px;
  word-break: break-word;
`;

export default Code;
