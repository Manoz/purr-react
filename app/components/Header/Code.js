import styled from 'styled-components';

const Code = styled.code`
  font-weight: 400;
  font-size: 87.5%;
  color: var(--secondaryColor);
  word-break: break-word;
`;

export default Code;
