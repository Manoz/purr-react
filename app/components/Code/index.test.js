import React from 'react';
import { shallow } from 'enzyme';

import Code from './index';

describe('<Code />', () => {
  it('should render an <code> tag', () => {
    const wrapper = shallow(<Code />);
    expect(wrapper.type()).toEqual('code');
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const wrapper = shallow(<Code id={id} />);
    expect(wrapper.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const wrapper = shallow(<Code attribute="test" />);
    expect(wrapper.prop('attribute')).toBeUndefined();
  });
});
