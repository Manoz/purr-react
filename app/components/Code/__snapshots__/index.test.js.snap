// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Code /> should render an <code> tag 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Code-ka9el6-0",
        "isStatic": true,
        "rules": Array [
          "padding:.2em .4em;margin:0;font-weight:400;font-size:88.5%;color:white;background-color:rgba(144,144,144,.09);border-radius:3px;word-break:break-word;",
        ],
      },
      "displayName": "Code",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Code-ka9el6-0",
      "target": "code",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
